sql/src/main/java/io/crate/analyze/MultiSourceSelect.java

```diff
@@ -118,24 +93,52 @@ public QuerySpec querySpec() {
         return querySpec;
     }
 
-    private RelationSplitter initSources(Map<QualifiedName, AnalyzedRelation> sources) {
-
-        RelationSplitter splitter = RelationSplitter.process(querySpec, sources.values());
-        for (Map.Entry<QualifiedName, AnalyzedRelation> entry : sources.entrySet()) {
+    public void initializeSources() {
+        splitter.process();
+        for (Map.Entry<QualifiedName, AnalyzedRelation> entry : originalSources.entrySet()) {
             QuerySpec spec = splitter.getSpec(entry.getValue());
             assert spec != null;
             Source source = new Source(entry.getValue(), spec);
-            this.sources.put(entry.getKey(), source);
+            sources.put(entry.getKey(), source);
         }
-        return splitter;
     }
 
     /**
      * Returns an orderBy containing only orderings which are not already pushed down to sources. The optional is only
-     * pressent if there are orderBySymbols left. The existing orderBy is returned if it
+     * present if there are orderBySymbols left. The existing orderBy is returned if it
      * there are no changes.
      */
     public Optional<OrderBy> remainingOrderBy() {
         return splitter.remainingOrderBy();
     }
+
+    public Map<QualifiedName, AnalyzedRelation> originalSources() {
```

Comment:
@dobe I agree, but we need to keep them to be able to [rebuild](https://github.com/crate/crate/pull/3883/files/3bfa4232f1f2315d09da74678d7e75edf2bfe690#diff-25d85154b2a99f88ee2b45cf3937e416R97) the MultiSourceSelect later on the RelationNormalizer. Do you think it makes sense to create a build method that receives the original MultiSourceSelect with other params to create the merged relation?